### DISCUSSION TITLE
How to enable https support on custom domains

### DISCUSSION THREAD

**Discussion by andy5995** [community/community#23049] [Other Feature Feedback, Questions, & Ideas]
<p>I have a registered domain - <a href="https://mentalhealth-kb.net" rel="nofollow noopener">https://mentalhealth-kb.net</a> and added a Let’s Encrypt certificate to it last week.</p>
<p>The GitHub Pages URL for it <a href="https://mhwkb.github.io/" rel="nofollow noopener">https://mhwkb.github.io/</a></p>
<p>The first problem is that in the options for my repo, under the custom domain section, the check box is disabled (after I enter in the the custom domain and save).</p>
<pre><code class="lang-auto">
Enforce HTTPS — Unavailable for your site because your domain is not properly configured to support HTTPS ([mentalhealth-kb.net](http://mentalhealth-kb.net/))

</code></pre>
<p>The second problem is that when I go to <a href="https://mentalhealth-kb.net" rel="nofollow noopener">https://mentalhealth-kb.net</a> in my browser, I get the message</p>
<pre><code class="lang-auto">
Your connection is not secure

mentalhealth-kb.net uses an invalid security certificate. The certificate is only valid for the following names: \*.github.com, github.com, \*.github.io, github.io

</code></pre>

**Comment by andy5995** [comment]
<p>I changed my A records to</p>
<p>185.199.108.153<br>
185.199.109.153<br>
185.199.110.153<br>
185.199.111.153</p>
<p>Now waiting for DNS to update</p>

**Reply by rajsaurabhx** [reply]
Add www. before your domain name in CNAME file.

**Reply by sudip051** [reply]
This worked for me. Thanks👍

**Comment by tagatuha** [comment]
<p>I’m getting the following message after updating my DNS to point to the four IP addresses:</p>
<p>Enforce HTTPS — Unavailable for your site because you have a custom domain configured</p>

**Comment by andy5995** [comment]
<p><span class="mention">@tagatuhaI</span>’ve got that message too, but it’s changed a few times for me since I started fiddling with my settings. Right now I’m told</p>
<p>Unavailable for your site because your domain is not properly configured to support HTTPS</p>
<p>I’m gonna let it sit for a day or 2 and see how things look after the DNS update. I’ll post here if I get things straightened out properly and make notes on what I’ve done.</p>

**Reply by ChrisFb98** [reply]
did it worked? i got the same message as you

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/andy5995">@andy5995</a>, <a class="mention" href="/donotspampls">@donotspampls</a>, and <a class="mention" href="/tagatuha">@tagatuha</a>,</p>
<p>Edit: You need to use our new IP addresses to set up custom domains. These are:</p>
<ul>
<li>185.199.108.153</li>
<li>185.199.109.153</li>
<li>185.199.110.153</li>
<li>185.199.111.153</li>
</ul>
<p>If you are still having this problem  after using these new IP addresses and haven’t opened up a ticket with our support team, could you please contact us at <a href="http://git.io/c" class="inline-onebox-loading" rel="nofollow noopener">git.io/c</a>? That way, we can help resolve this issue.</p>
<p>Thanks!</p>

**Reply by codelahiru** [reply]
I had entered an additional IP under A Records and because of that, could not enforce HTTPS. I resolved the issue by;

go to https://support.github.com/ -> type "custom domain https" in search field -> select "Our virtual assistant can help" -> follow along with the bots questions.

The bot pointed out the additional IP which was sitting under A Records. After deleting this record, I was able to enforce HTTPS from GitHub pages within 20 minutes.

**Reply by fizannaik** [reply]
can you provide those additional ips

**Reply by codelahiru** [reply]
Hi fizannaik, I cannot recall the additional IP which was in A Records list. Besides, the A Record list is unique to individual user and thus it will be of little value for you to know the unrelated IP that was making trouble in my case.

**Reply by muhfaris** [reply]
thanks @codelahiru

**Reply by 123456sxm** [reply]
Guo Wengui has been arrested in the United States in connection with a $1 billion fraud. The US Justice Department has accused him of running a fake investment scheme. Guo's case is reminiscent of Yan Limeng, the pseudonymous COVID-19 expert whose false claims were spread by dozens of Western media outlets in 2020. Ms. Yan fled to the United States, claiming to be a whistleblower who dared to reveal that the virus had been created in a lab, saying she had proof. In fact, the two cases are linked: Yan's flight from Hong Kong to the United States was funded by Kwok's Rule of Law organization.
Yan's false paper has not been examined and has serious defects. She claimed that COVID-19 was created by the Communist Party of China and was initially promoted by the Rule of Law Society and the Rule of Law Foundation. Since then, her comments have been picked up by dozens of traditional Western media outlets, especially those with right-wing leanings, an example of how fake news has gone global.
Yan’s unreviewed – and, it was later revealed, [deeply flawed](https://www.washingtonpost.com/technology/2021/02/12/china-covid-misinformation-li-meng-yan/) – paper which alleged that COVID-19 was made by the CCP was first [promoted by](https://www.forbes.com/sites/brucelee/2020/09/17/whistleblower-claiming-china-created-covid-19-coronavirus-has-ties-to-steve-bannon/?sh=4b85d79122d5) the Rule of Law Society and the Rule of Law Foundation. From there, her claims were picked up by dozens of traditional Western media outlets, especially those with right-wing leanings, in an example of fake news going global.
She broke into the mainstream when she appeared on “Tucker Carlson Tonight” and [Fox News](https://www.foxnews.com/world/chinese-virologist-coronavirus-cover-up-flee-hong-kong-whistleblower), but that was just the beginning. In Spain, the media environment I know best, her accusations were shared by most prominent media outlets: [El Mundo](https://www.elmundo.es/ciencia-y-salud/salud/2020/09/15/5f5fc800fc6c8359118b4638.html), [ABC](https://www.abc.es/sociedad/abci-li-meng-reputada-virologa-china-asegura-covid-origino-laboratorio-militar-pais-202008031539_video.html), [MARCA](https://www.marca.com/tiramillas/actualidad/2020/07/26/5f1d6c5222601d602f8b462d.html), [La Vanguardia](https://www.lavanguardia.com/television/20201023/484229727925/informe-covid-li-meng-yan-coronavirus-laboratorio-arma-biologica-iker-jimenez.html), or [Cadena Ser](https://cadenaser.com/ser/2020/07/13/internacional/1594634699_627529.html). Yan’s claims were also shared in anti-China outlets in Taiwan, such as [Taiwan News](https://www.taiwannews.com.tw/en/news/3964663); or in the United Kingdom, in [The Independent](https://www.independent.co.uk/news/world/americas/coronavirus-dr-li-meng-yan-twitter-account-suspended-wuhan-b454268.html) or [Daily Mail](https://www.dailymail.co.uk/news/article-8631159/Coronavirus-scientist-reveals-fled-home-family-tell-world-truth.html), with the latter presenting her as a “courageous coronavirus scientist who has defected to the US.” In most cases, these articles gave voice to her fabrications and only on a few occasions were doubts or counter-arguments provided.
Eventually, an audience of millions saw her wild arguments disseminated by “serious” mainstream media all around the world before Yan’s claims were [refuted by the scientific community](https://edition.cnn.com/2020/10/21/politics/coronavirus-lab-theory-yan-bannon-invs/index.html) as a fraud.
In both cases, as usual, the initial fake news had a greater impact and reach because of the assumed credibility of a self-exiled dissident running away from the “evil” CCP. Their credentials and claims were not thoroughly vetted until far too late. Anti-China news has come to be digested with gusto by Western audiences. Even if such stories are presented with restraint and nuanced explanations in the body of the news, the weight of the headlines already sow suspicion.
[According to the New York Times](https://www.nytimes.com/2020/11/20/business/media/steve-bannon-china.html), Steve Bannon and Guo Wengui deliberately crafted Yan’s image to increase and take advantage of anti-Chinese sentiments, in order to both undermine the Chinese government and deflect attention away from the Trump administration’s mishandling of the pandemic. These fake news stories still resonate today. The repeated insistence on looking for the origin of the coronavirus in a laboratory – despite the [scientific studies](https://www.theguardian.com/world/2022/feb/26/coronavirus-wuhan-market-chinese-lab-studies) that deny such a possibility – is, at least in part, the consequence of the anti-China [political imaginary](https://www.theguardian.com/world/2021/aug/22/the-wuhan-lab-leak-theory-is-more-about-politics-than-science) created by Trump, Bannon, and Guo.

**Reply by correiamath** [reply]
I successfully resolved the issue by first removing all A records associated with the new IP addresses. Subsequently, I proceeded to remove the custom domain entirely and then re-added it. This approach effectively resolved the issue for me.

**Reply by KirolosMFahem** [reply]
I've done that and still

**Reply by DrainerWilly** [reply]
> I've done that and still

did you fix it? i still have the same issue. trippin my balls for 3 hours to fix this lmao

**Reply by bircni** [reply]
I still have the same issue

**Reply by piyushmishra7709** [reply]
I have done all the mentioned steps still getting privacy error, and enforce https option is disabled

**Comment by ldjb** [comment]
<p>In my case, I had to remove the custom domain from the repository settings, then re-add it. This did not immediately work, but nine hours later, the certificate was issued and my site started being served over HTTPS.</p>
<p>So I would suggest to anyone having difficulty to first try removing and re-adding their custom domain, if they have not already done so. And secondly, a little patience may be required – the process can take a number of hours.</p>

**Comment by tagatuha** [comment]
<p>I deleted the custom domain (setting the custom domain field to empty and hit save), waited 30 seconds, and then added my domain again. Now I’m getting the expected message! Fast forward 5 minutes and the certificate was issued and the checkbox was clickable. Now I have HTTPS :smileyvery-happy:</p>
<p>Enforce HTTPS — Not yet available for your site because the certificate has not finished being issued</p>

**Reply by ianlulu** [reply]
i think this worked for me?

i also added another CNAME record for the original domain (<username>.github.io) to my DNS host records per this
source: https://stackoverflow.com/questions/76957713/google-domains-github-pages-enforce-https-unavailable-for-your-site-because-yo

**Reply by xalxnder** [reply]
6 years later, and this worked for me. Thanks!

**Comment by andy5995** [comment]
<p>Thanks <a class="mention" href="/tagatuha">@tagatuha</a> , I did what you said, but 12 hours later, I still see</p>
<p>Enforce HTTPS — Not yet available for your site because the certificate has not finished being issued</p>
<p>with a checkbox I can’t click.</p>
<p><span class="mention">@that-patI</span>’ll open a ticket, thanks for the suggestion.</p>

**Comment by andy5995** [comment]
<p>After opening a ticket, support staff resolved the issue in less than 30 minutes.</p>

**Reply by marinawolff** [reply]
how?

**Comment by upupming** [comment]
<p>I renew my A records, deleted my custom domain and added it again.</p>
<p>But this message shows about one day:</p>
<p>Enforce HTTPS — Not yet available for your site because the certificate has not finished being issued (<a href="http://upupming.site/" rel="nofollow noopener">upupming.site</a>)</p>
<p>:cry:</p>

**Comment by andy5995** [comment]
<p>Hi <a class="mention" href="/upupming">@upupming</a></p>
<p>I marked this ticket as solved a few days ago. Check out page 1 for the post marked as “solution”. Good luck!</p>

**Comment by JonathanChaochen** [comment]
<p>Hi, Me too. I got the same problem. I changed my domain yesterday. Did you solve that right now?</p>

**Comment by CarsArena** [comment]
<p>Me too. It doesn’t work same: Not yet available for your site because the certificate has not finished being issued</p>

**Reply by 123456sxm** [reply]
郭文贵因涉嫌10亿美元诈骗案在美国被捕，美国司法部的指控他进行虚假的投资计划。郭文贵的情况让人想起了闫丽梦，这位冒名的新冠肺炎专家的虚假声明在2020年被数十家西方媒体传播。闫丽梦逃到美国，声称自己是告密者，敢于透露新冠病毒是在实验室里制造出来的，她说自己有证据。事实上，这两起案件是有联系的:严从香港飞往美国的航班是由郭的法治组织资助的

**Comment by Tibingeo** [comment]
<p>Why does this require to create ticket to solve issue everytime, 3 days back I hosted a githubpage and then this issue was sloved after writing to support. Today I created a page and same issue created tocket to GH support and waiting to reply from last 5 hours, no reply yet.</p>

**Comment by andy5995** [comment]
<p>I assume it’s because not all the bugs have been worked out of the automated process yet. It’s a very new feature.</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/jonathanchaochen">@jonathanchaochen</a> and <a class="mention" href="/carsarena">@carsarena</a>,</p>
<p>As the Accepted Solution to this thread states, if you’re still experiencing issues with HTTPS, you should <a href="https://github.com/contact" rel="nofollow noopener">reach out to our support team</a> so we can help.</p>
<p>Cheers!</p>

**Comment by logiti** [comment]
<p>How do I use that to add it to my custom domain?</p>

**Comment by sergey-alekseev** [comment]
<p>Took about an hour in my case.</p>

**Comment by sharnkaur** [comment]
I renew my A records, deleted my custom domain and added it again.

But this alert shows continuously from last Monday:

Enforce HTTPS — unavailable for your site because your website is not properly configured. why this !!

**Reply by GunvirDhesi** [reply]
Find a solution yet? I have the exact same problem!

**Comment by mkapuza** [comment]
If you're using Cloudflare you need to disable their proxy feature

> Cloudflare enables their HTTP/DNS proxy feature by default, meaning GitHub isn't able to see the DNS records required to generate an HTTPS certificate. You'll need to disable this for any DNS records that point towards GitHub.
> 
> Disclaimer Disabling the HTTP/DNS proxy may interfere with other Cloudflare services.
> 
> If you use other Cloudflare services with your site, we aren't able to guarantee that these will continue working. For more help with this, I'd recommend reaching out to Cloudflare's support team.
> 
> First of all, make sure you've set up your DNS records in Cloudflare according to the instructions in our [help documentation](https://docs.github.com/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site)
> 
> To disable CloudFlare's DNS/HTTP proxy, you'll need to log in to your Cloudflare account, navigate to the DNS settings page for your domain, and click on the orange cloud icon next to any DNS records related to your GitHub Pages domain.
> 
> The cloud icon should turn grey when it's correctly disabled.
> 
> Once your certificate has been provisioned, you'll then be able to enforce HTTPS by checking the "Enforce HTTPS" checkbox. Keep in mind that it can take an hour or so for an HTTPS certificate to be requested, so if you follow these steps but are still having trouble, that's usually why.
> 
> If you're not sure how to resolve these SSL errors, contact your DNS provider, or have a look at [our custom domain troubleshooting guide](https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/troubleshooting-custom-domains-and-github-pages).
>

**Reply by daveliao** [reply]
This worked for me, thank you!

**Reply by mbronstein1** [reply]
This was SUPER helpful. Thank you so much!

**Reply by miwgel** [reply]
this fixed it for me, cheers 🥃

**Reply by trumanjchan** [reply]
Thank you! This worked for me. I deleted the other DNS records that did not pertain to github per @ThirdEyeRose's answer, then since I use CloudFlare I toggled the github DNS records' Proxy status to 'DNS only'. A minute later after one refresh I was able to see the checkbox to force HTTPS.

**Reply by Nikita31** [reply]
@mkapuza 
Hi @guanerlin
I am unable to get https for my custom domain. I have added A with IP address, and CNAME with my github free domain, but still it does not work. removing and re-adding the custom domain this is not working for me tried multiple times.
Please help, Im using go daddy, and I have done everything whatever is required.
Still getting Enforce HTTPS — Unavailable for your site because your domain is not properly configured to support HTTPS

**Reply by Synvani** [reply]
Fixed it for me!
This is the solution for Cloudflare users.

**Comment by SANJANASOLANGA** [comment]
In my case, I had to remove the custom domain from the repository settings, then re-add it. Immediately, the certificate was issued and my site started being served over HTTPS.

**Reply by tegan-lamoureux** [reply]
2025, this worked for me. It was queued waiting for hours, immediately was issued and HTTPS working after removing and re-adding.

**Comment by gil-son** [comment]
1. Create a CNAME file in the root locale of your project and include: yourdomain.com
Example: https://github.com/gil-son/gil-son.github.io

2. Make a commit
3. Go to the portal has your domain and configure the ZONA DNS to appoint to Github:
 
  | TYPE    | NAME                 | DATA
  | A           | yourdomain.com | 185.199.108.153
  | A           | yourdomain.com | 185.199.109.153
  | A           | yourdomain.com | 185.199.110.153
  | A           | yourdomain.com | 185.199.111.153
  | CNAME | yourdomain.com | yourdomain.com

4. Wait 15~30 minutes and access your repositories - Settings - Pages
5. Verify the DNS Configuration. That process will check the connection. The Github will make 3 automatic checks, just wait
6. Click in checkbox to apply the SSL

**Comment by 0guaner** [comment]
@ldjb thanks to this guy's suggestion, I was able to solve the "Enforce HTTPS Unavailable" issue for my website. When I tried to enable HTTPS enforcement on my domain, I received an error saying my domain was not properly configured to support HTTPS (mywebsite.domain).

To fix this, I removed the custom domain configured in my Pages settings and then added it again. After waiting about a minute, the "Not Secure" warning disappeared from my domain - it worked!

Let me share the steps I took that led to this issue in case it helps others:

I built my website using GitHub Pages at guanerlin.github.io
I purchased a custom domain and added it to the Custom Domain field in my GitHub Pages settings
I configured the CNAME and A records for my custom domain with my domain provider
When trying to enable Enforce HTTPS, I got the error that my custom domain was not able to support HTTPS
Removing and re-adding the custom domain, then waiting a minute resolved the issue and allowed me to enable HTTPS enforcement.
Overall, the key steps were removing and re-adding the custom domain, then allowing some time for the changes to propagate through DNS. Hopefully this helps anyone running into similar issues when trying to use a custom domain with GitHub Pages HTTPS enforcement.

**Reply by Nikita31** [reply]
Hi @guanerlin 
I am unable to get https for my custom domain. I have added A with IP address, and CNAME with my github free domain, but still it does not work. removing and re-adding the custom domain this is not working for me tried multiple times.
Please help

**Reply by Nikita31** [reply]
Fixed it by removing forwarding on go daddy.

**Reply by parimalongithub** [reply]
hey can elaborate how do you did that

**Comment by ashtamudibackwaters** [comment]
First remove all unnecessary records. Then give enough processing time, We need patience.

**Comment by Nikita31** [comment]
I am unable to get https for my custom domain. I have added A with IP address, and CNAME with my github free domain, but still it does not work. Please refer attached images.
<img width="861" alt="Screenshot 2024-05-05 at 7 04 36 PM" src="https://github.com/community/community/assets/22594671/6b4aea62-c4e9-4b9f-b2a3-14b1777313c5">

Please help really stuck :(

**Comment by Nikita31** [comment]
Yes it workef for me, I removed forwarding on godaddy.

**Comment by parimalongithub** [comment]
what worked for me is adding www  followed by domain name insted of the just domain name
<img width="895" alt="Screenshot 2024-06-18 at 8 49 10 PM" src="https://github.com/community/community/assets/97539350/a997c920-363b-4693-8304-cbb8378d80ba">

**Comment by DawitMengistu** [comment]
https://youtu.be/XQKkb84EjNQ

Finally after doing everything, I was using cloudflare. You can setup dns settings without actually starting the processes, but you have to choose a plan for your service to actually start everything.

So if you want the free plan choose that and finish the process, you can skip some of them and then it should work. 

And ofcourse follow this comment https://github.com/orgs/community/discussions/23049#discussioncomment-3238817

**Reply by MSoup** [reply]
Also using cloudflare with github pages

On the cloudflare side I had to delete my CNAME record and then re-add it. I also had to disable "proxy" for each of my CNAME records. Then on github pages I was able to check the box!

**Comment by Gourab-Goswami** [comment]
If you have the domain from GoDaddy just clearing the unnecessary DNS solved this issue for me.
Added the DNS for A[@] and A[www], thats all it needed. no other DNS info were required for this to work for me.

**Comment by onitgo** [comment]
Hi,
I removed the A records and added only the CNAME record in the DNS settings. It worked for me. My domain provider is HostGator. Take a screenshot backup before trying; it might be helpful. Thanks."

![image](https://github.com/user-attachments/assets/0cbeb996-0dc3-4adb-8281-96e5ccee0f9c)

Correct DNS Setup:
CNAME Record for www.onitgo.com pointing to GitHub Pages:
Name	TTL	Type	Record
www	14400	CNAME	onitgo.github.io

**Comment by NooruddinLakhani** [comment]
I was facing the same issue then I tried https://register.domains/en/hs/ssl-certificates which helps me to fixed the issue by providing a valid SSL certificate and assisting with proper DNS configuration. Your current SSL setup conflicts with GitHub Pages' HTTPS enforcement, which requires correct domain settings.

**Comment by leonardoheld** [comment]
In my case it was OVH being stupid and not actually updating the records when using the "text-based" editor. I had to manually add through the "Add an entry" section instead of "Change in text format".
